plugins/generator-1.21.1/neoforge-1.21.1/templates/livingentity/livingentity.java.ftl

```diff
@@ -976,5 +1031,74 @@ public class ${name}Entity extends ${extendsClass} <#if data.ranged>implements R
 		return builder;
 	}
 
+	<#if data.sensitiveToVibration>
+	private class VibrationUser implements VibrationSystem.User {
+
+		private final ${name}Entity entity = ${name}Entity.this;
+		private final int x = this.entity.getOnPos().getX();

```

Comment:
VibrationUser is only created on spawn, so those coordinates will remain fixed to the spawn position, which is not ok for the procedures. I believe you should directly access location of the entity in the trigger code itself inside "@procedureCode"